docs/machine-learning/how-to-guides/prepare-data-ml-net.md

```diff
@@ -0,0 +1,290 @@
+
+```csharp
+// Define replacement estimator
+var replacementEstimator = mlContext.Transforms.ReplaceMissingValues("Price", replacementMode: MissingValueReplacingEstimator.ReplacementMode.Mean);
+
+// Build transformer to perform operations
+ITransformer replacementTransformer = replacementEstimator.Fit(data);
+
+// Transform data
+IDataView transformedData = replacementTransformer.Transform(data);
+```
+
+ML.NET supports various [replacement modes](xref:Microsoft.ML.Transforms.MissingValueReplacingEstimator.ReplacementMode). The sample above uses the `Mean` replacement mode which will fill in the missing value with that column's average value. The replacement
+'s result fills in the `Price` property for the last element in our data with 200,000 since it's the average of 100,000 and 300,000. 
+
+## Filter data
+
+Sometimes, not all data in a dataset is relevant for analysis. An approach to remove irrelevant data is filtering.
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+HomeData[] homeDataList = new HomeData[]
+{
+    new HomeData
+    {
+        NumberOfBedrooms=1f,
+        Price=100000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms=2f,
+        Price=300000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms=6f,
+        Price=600000f
+    }
+};
+```
+
+To filter data based on the value of a column, use the [`FilterRowsByColumn`](xref:Microsoft.ML.DataOperationsCatalog.FilterRowsByColumn*) method.
+
+```csharp
+// Apply filter
+IDataView filteredData = mlContext.Data.FilterRowsByColumn(data, "Price", lowerBound: 200000, upperBound: 1000000);
+```
+
+The sample above takes rows in the dataset with a price between 200000 and 1000000. The result of applying this filter would return only the last two rows in the data and exclude the first row because its price is 100000 and not between the specified range.
+
+## Use normalizers
+
+[Normalization](https://en.wikipedia.org/wiki/Feature_scaling) is a data pre-processing technique used to standardize features that are not on the same scale. For example, the ranges for values like age and income vary significantly with age generally being in the range of 0-100 and income generally being in the range of zero to thousands. Visit the [transforms page](../resources/transforms.md) for a more detailed list and description of normalization transforms. 
+
+### Min-Max normalization
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+HomeData[] homeDataList = new HomeData[]
+{
+    new HomeData
+    {
+        NumberOfBedrooms = 2f,
+        Price = 200000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms = 1f,
+        Price = 100000f
+    }
+};
+```
+
+Normalize the data using min-max normalization using the [`NormalizeMinMax`](xref:Microsoft.ML.NormalizationCatalog.NormalizeMinMax*) method.
```

Comment:
>Normalize [](start = 0, length = 9)

Maybe we should say why normalizer the data. Typically normalization is done so the algorithm converges quicker. 
This is typically needed for linear trainers. 

[https://docs.microsoft.com/en-us/azure/machine-learning/studio-module-reference/normalize-data](https://docs.microsoft.com/en-us/azure/machine-learning/studio-module-reference/normalize-data)